### ISSUE TITLE
MD013: LSP diagnostic persist after formatting with `semantic-line-breaks`

### ISSUE THREAD

**Issue by tahv** [rvben/rumdl#459]
### rumdl version

0.1.25

### Installation method

Pre-built binary

### Bug description

Hi,

When running `rumdl` LSP (in neovim), the wargning diagnostic for `MD013` with  `reflow-mode = "semantic-line-breaks"` show up even after running the formatter.

### Steps to reproduce

1. Create a fime `repro.md`:

```markdown
# Title

Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Nullam vehicula commodo lobortis.
Donec a venenatis lorem.
```

### Expected behavior

No LSP warning if text if formatted properly.

### Actual behavior

LSP warning:

<img width="915" height="165" alt="Image" src="https://github.com/user-attachments/assets/4ccebaea-56c5-46ec-8ace-98b7dbfa7f6d" />

No warning from `rumdl check`:

```console
$ rumdl check repro.md 
                                                                                                                                                                       
Success: No issues found in 1 file (37ms)

```

Output of `checkhealth vim.lsp` for `rumdl`:

```text
- rumdl:
  - capabilities: {
      textDocument = {
        completion = {
          completionItem = {
            commitCharactersSupport = false,
            deprecatedSupport = true,
            documentationFormat = { "markdown", "plaintext" },
            insertReplaceSupport = true,
            insertTextModeSupport = {
              valueSet = { 1 }
            },
            labelDetailsSupport = true,
            preselectSupport = false,
            resolveSupport = {
              properties = { "documentation", "detail", "additionalTextEdits", "command", "data" }
            },
            snippetSupport = true,
            tagSupport = {
              valueSet = { 1 }
            }
          },
          completionList = {
            itemDefaults = { "commitCharacters", "editRange", "insertTextFormat", "insertTextMode", "data" }
          },
          contextSupport = true,
          insertTextMode = 1
        }
      }
    }
  - cmd: { "rumdl", "server" }
  - filetypes: markdown
  - root_markers: { ".git" }
```

### Configuration

```toml
[tool.rumdl.MD013]
line-length = 80
reflow = true
reflow-mode = "semantic-line-breaks"
```

### Operating system

Windows

**Comment by rvben** [comment]
Thanks for reporting @tahv, fixed in v0.1.27. The reflow logic was hardcoding LF line endings, so editors sending CRLF content via LSP would always see false positives (the reflowed text never matched the original). Line ending style is now detected and preserved.
